client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -581,6 +588,22 @@ export class CheckoutThankYou extends Component<
 				pageContent = <DomainOnlyThankYou purchases={ purchases } receiptId={ receiptId } />;
 			} else if ( purchases.length === 1 && isPlan( purchases[ 0 ] ) ) {
 				pageContent = <PlanOnlyThankYou primaryPurchase={ purchases[ 0 ] } />;
+			} else if ( wasTitanEmailOnlyProduct ) {
+				pageContent = (
+					<TitanSetUpThankYou
+						domainName={ purchases[ 0 ].meta }
+						emailAddress={ email }
+						isDomainOnlySite={ this.props.domainOnlySiteFlow }
+					/>
+				);
+			} else if ( isTitanWithoutMailboxes( selectedFeature ) && domainPurchase ) {
+				// To reproduce it: https://github.com/Automattic/wp-calypso/pull/87098#titan-wo-mailboxes
+				pageContent = (
+					<TitanSetUpThankYou
+						domainName={ domainPurchase.meta }
+						isDomainOnlySite={ this.props.domainOnlySiteFlow }
+					/>
+				);
```

Comment:
```suggestion
			} else if ( isTitanWithoutMailboxes( selectedFeature ) ) {
				// Users may purchase Titan subscription without specifying the mailbox name using
				// the onboard with email flow (https://wordpress.com/start/onboarding-with-email/mailbox-domain)
				const titanPurchase = purchases.find( ( purchase ) => isTitanMail( purchase ) );
			
				pageContent = titanPurchase && (
					<TitanSetUpThankYou
						domainName={ titanPurchase.meta }
						isDomainOnlySite={ this.props.domainOnlySiteFlow }
					/>
				);
```

The suggested revisions address two things:
* I think it's clearer to describe the scenario rather than pointing to a PR.
* We can get the domain from the titan purchase instead of getting the domain purchase.

Reply:
IMO the current approach a bit more clear, let me elaborate:
1) I know that the flow is buying domain and titan, so the condition about "buying domain" is more clear, when I see only the condition about "titan purchase". As "domain and titan w/o mailboxes", instead of "titan purchase and titan w/o mailboxes".
2) Just personal - it's more clear at a glance when we have one condition, instead of two, and the second one is located inside the first condition, but it's just personal.
3) `domainName={ domainPurchase.meta }` looks a bit more readable and clear, than `domainName={ titanPurchase.meta }`.

All my arguments are obviously and not strong, so if you like much more `titanPurchase `, I am totally ok to go with it. Waiting for your final decision, taking into account my points above.

Reply:
>I know that the flow is buying domain and titan, so the condition about "buying domain" is more clear, when I see only the condition about "titan purchase". As "domain and titan w/o mailboxes", instead of "titan purchase and titan w/o mailboxes".

I can agree with this. It's true that in our current set up, the only way users would end up purchasing a Titan subscription without defining a name for the mailbox is when they purchase a domain AND email subscription. If you prefer keeping `domainPurchase`, that's fine with me! But I would still suggest cleaning up the comment so we describe the scenario rather than pointing to a PR :)

Reply:
>But I would still suggest cleaning up the comment so we describe the scenario rather than pointing to a PR :)

Yeah, I have already changed both comments :)